Gym/FrozenLake.swift

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Python
+import TensorFlow
+
+let np = Python.import("numpy")
+let gym = Python.import("gym")
+
+/// Solves the FrozenLake RL problem via Q-learning. This model does not use a
+/// neural net, and instead demonstrates Swift host-side numeric processing as
+/// well as Python integration.
+
+let discountRate: Float = 0.9
+let learningRate: Float = 0.2
+let testEpisodeCount = 20
+
+typealias State = Int
+typealias Action = Int
+
+// Force unwrapping with ! does not provide source location when unwrapping
+// nil, so we instead make a util function for debuggability.
+fileprivate extension Optional {
+  func unwrapped(file: StaticString = #file, line: UInt = #line) -> Wrapped {
+    guard let unwrapped = self else {
+      fatalError("Value is nil", file: file, line: line)
+    }
+    return unwrapped
+  }
+}
+
+// This struct is defined so that `StateAction` can be a dictionary key
+// type. Swift tuples cannot be dictionary key types.
+struct StateAction : Hashable {
```

Comment:
Remove space before colon.
```suggestion
struct StateAction: Hashable {
```

Reply:
done